ctrd/image.go

```diff
@@ -160,7 +172,7 @@ outer:
 				}
 				// update status of active entries!
 				for _, active := range active {
-					progresses[active.Ref] = ProgressInfo{
+					progresses[active.Ref] = &ProgressInfo{
```

Comment:
Actually I do not like this block. It is so long and so hard to understand without annotations. Maybe in a follow-up, we need to refactor this a little bit to make this more readable.